server/src/MessageHandler.ts

```diff
@@ -26,12 +60,65 @@ function getWordAtPosition(document: TextDocument, position: Position): string {
   return currentLine.slice(beginning, end + position.character)
 }
 
+export class MyBlock {
+  type: string
+  start: Position
+  end: Position
+  name: string
+
+  constructor(type: string, start: Position, end: Position, name: string) {
+    this.type = type
+    this.start = start
+    this.end = end
+    this.name = name
+  }
+}
+
+function getBlockAtPosition(
+  line: number,
+  document: TextDocument,
+): MyBlock | undefined {
+  if (document) {
+    let blockType = ''
+    let blockName = ''
+    let blockStart: Position = Position.create(0, 0)
+    let blockEnd: Position = Position.create(0, 0)
+    // get block beginning
+    for (let _i = line; _i >= 0; _i--) {
```

Comment:
**nitpick:** any reason for the `_i`? It seems like we should go with camelCase or snake_case, but not both.

Reply:
no not really, I'll change that